docs/reference/glossary.asciidoc

```diff
@@ -105,6 +115,20 @@
   +
   See also <<glossary-routing,routing>>
 
+[[glossary-query]] query ::
+
+  A query is the basic component of a search. A search can be defined by one or more queries 
+  which can be mixed and matched in endless combinations. The term Query refers to all queries 
```

Comment:
I'd scratch this statement, I think of "Query" as a superset of both "scoring queries" and filters, so I wouldn't introduce a distinction here.

Reply:
Please, I will like to clarify this. So you mean whenever the term **Query** is used it can either mean "filters" or "scoring queries". So, unless it is clarified it can be either (or I guess, even both).

Reply:
Others might disagree, but my understanding is that we only have "queries" in elasticsearch now (there used to be special filters in earlier versions). Some queries, if used in a "filtering context" as described in the link mentioned above, are also refered to as "filters" or "non-scoring queries" as oposed to "scoring queries" (when the queries are producing scores for documents). Whether something is a filter or not depends on their location. e.g. a `term_query` is generally speaking alway a query, but when it is used e.g. in a `boolean` queries `filter` clause it is also refered to as a filter in our documentation.

Reply:
The above explanation might be a bit verbose for the glossary however, so making it a bit more succinct would be the  goal here.